networkx/algorithms/matching.py

```diff
@@ -1150,3 +1149,226 @@ def verifyOptimum():
         verifyOptimum()
 
     return matching_dict_to_set(mate)
+
+
+def count_planar_perfect_matchings(G: nx.Graph):
+    """Counts the number of perfect matchings using the
+    Fisher-Kasteleyn-Temperley (FKT) algorithm. `G` must be planar in order to
+    use the FKT algorithm.
+
+    If `G` is a weighted graph, each matching is counted with weight equal to
+    the product of weights of edges included in the matching. Edges without
+    weights default to having weight 1.
+
+    Technically, the FKT algorithm only returns the absolute value of the sum
+    of perfect matchings. Generally one can tell for other reasons whether the
+    sum of perfect matchings should be positive or negative for their problem.
+
+    If `G` is a directed graph, this ignores the edge directions and considers
+    `G` as undirected.
+
+    Raises NetworkXError if `G` is not planar.
+
+    Parameters
+    ----------
+    G : NetworkX Graph
+
+    Returns
+    ----------
+    float
+        Sum (possibly weighted) of perfect matchings in `G`.
+
+    Raises
+    ----------
+    NetworkXError
+        If `G` is not planar.
+
+    References
+    ----------
+    .. [1] P. W. Kasteleyn:
+        Dimer Statistics and Phase Transitions
+        1963
+        https://doi.org/10.1063/1.1703953
+    """
+
+    # If we're given a directed graph, we just convert it to undirected.
+    if isinstance(G, nx.DiGraph):
```

Comment:
By using a nicely named method the comment is no longer needed... (?)
```suggestion
    if G.is_directed():
```